### PR TITLE
Dump-only fields must be skipped if 'dump=False' is passed

### PR DESCRIPTION
**Pull request by frol**

### CODE REVIEW

File: apispec/utils.py [hunk 1]
```diff
@@ -98,4 +98,4 @@ def validate_swagger(spec):
                 stderr=subprocess.STDOUT,
             )
         except subprocess.CalledProcessError as error:
-            raise exceptions.SwaggerError(error.output.decode('utf-8'))
+            raise exceptions.PluginError(error.output.decode('utf-8'))
```
Comment (sloria) [inline apispec/utils.py]: Ah, good catch. I think the proper fix is to add `SwaggerError` to `exceptions`. A `PluginError` is raised for plugin-related code.

**Comment by sloria** [comment]
Thanks! This looks good. Once the SwaggerError is added, this should be good to merge.

File: apispec/utils.py (hunk 1 above)
Comment (frol) [inline apispec/utils.py]: Please, go ahead and fix this the way you like it. I have only fixed it so the tests would pass (I don't know how they did pass before).

Comment (frol) [inline apispec/utils.py]: I'm just not really sure if swagger mentions should be somewhere but in `ext/marshmallow/swagger.py`.

**Comment by frol** [comment]
@sloria Added SwaggerError as you pointed out.

**Comment by sloria** [comment]
Thanks!

**Pull request merged**